src/main/scala/com/ubirch/util/mongo/connection/Connection.scala

```diff
@@ -0,0 +1,68 @@
+package com.ubirch.util.mongo.connection
+
+import com.typesafe.scalalogging.LazyLogging
+import com.ubirch.util.mongo.config.{ MongoConfig, MongoConfigKeys }
+import com.ubirch.util.mongo.connection.Exceptions._
+import reactivemongo.api.MongoConnection.ParsedURI
+import reactivemongo.api.{ AsyncDriver, MongoConnection }
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+import scala.util.{ Failure, Success, Try }
+
+trait ConnectionTrait {
+
+  def driver: AsyncDriver
+
+  def uri: String
+
+  def parsedUri: Future[ParsedURI]
+
+  def conn: Future[MongoConnection]
+
+  def close(): Unit = driver.close()
+
+  def closeLogical(): Unit = {
+    conn.map(_.close()(2.seconds))
+  }
+
+  def connIsActive: Future[Boolean] = conn.map(_.active)
+
+}
+
+class Connection private (configPrefix: String) extends ConnectionTrait {
+
+  val driver: AsyncDriver = new reactivemongo.api.AsyncDriver
+  val uri: String = MongoConfig.hosts(configPrefix)
+  val parsedUri: Future[ParsedURI] = MongoConnection.fromString(uri)
+  val conn: Future[MongoConnection] = parsedUri.flatMap((uri: ParsedURI) => driver.connect(uri))
+
+}
+
+object Connection extends LazyLogging {
+
+  private var connection: Option[Connection] = None
```

Comment:
FYI: Atomic access such as using `AtomicAny` can be more efficient rather than synchornized
but, I guess it doesn't affect so much here
https://stackoverflow.com/questions/11670687/when-is-atomicinteger-preferrable-over-synchronized

Reply:
thanks. reduced accessing synchronized part..